californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -199,7 +200,7 @@ public void sendRequest(final Exchange exchange, final Request request) {
 		if (request.getOptions().hasObserve() && request.getOptions().getObserve() == 0 && (!request.getOptions().hasBlock2()
 				|| request.getOptions().getBlock2().getNum() == 0 && !request.getOptions().getBlock2().isM())) {
 			// add request to the store
-			observationStore.add(new Observation(request, null));
+			List<Observation> observationRemoved = observationStore.add(new Observation(request, null));
```

Comment:
When we say there can be at most only one observe per resource, should not add(Observation) return one Observation instead of a List?